test/echo.js

```diff
@@ -2,137 +2,161 @@ import test from 'ava';
 
 import shell from '..';
 import utils from './utils/utils';
+import mocks from './utils/mocks';
 
 shell.config.silent = true;
 
 test.beforeEach(t => {
   t.context.tmp = utils.getTempDir();
+  mocks.init();
 });
 
 test.afterEach.always(t => {
   shell.rm('-rf', t.context.tmp);
+  mocks.restore();
 });
 
 //
 // Valids
 //
 
-test.cb('simple test with defaults', t => {
-  const script = 'require(\'../global.js\'); echo("hello", "world");';
-  utils.runScript(script, (err, stdout, stderr) => {
-    t.falsy(err);
-    t.is(stdout, 'hello world\n');
-    t.is(stderr, '');
-    t.end();
-  });
+test('simple test with defaults', t => {
+  const result = shell.echo('hello', 'world');
+  const stdout = mocks.stdout();
+  const stderr = mocks.stderr();
+  t.falsy(shell.error());
+  t.is(result.code, 0);
+  t.is(stdout, 'hello world\n');
+  t.is(stderr, '');
 });
 
-test.cb('allow arguments to begin with a hyphen', t => {
+test('allow arguments to begin with a hyphen', t => {
   // see issue #20
-  const script = 'require(\'../global.js\'); echo("-asdf", "111");';
-  utils.runScript(script, (err, stdout, stderr) => {
-    t.falsy(err);
-    t.is(stdout, '-asdf 111\n');
-    t.is(stderr, '');
-    t.end();
-  });
+  const result = shell.echo('-asdf', '111');
+  const stdout = mocks.stdout();
+  const stderr = mocks.stderr();
+  t.falsy(shell.error());
+  t.is(result.code, 1);
+  t.is(stdout, '-asdf 111\n');
+  t.is(stderr, '');
 });
 
-test.cb("using null as an explicit argument doesn't crash the function", t => {
-  const script = 'require(\'../global.js\'); echo(null);';
-  utils.runScript(script, (err, stdout, stderr) => {
-    t.falsy(err);
-    t.is(stdout, 'null\n');
-    t.is(stderr, '');
-    t.end();
-  });
+test("using null as an explicit argument doesn't crash the function", t => {
+  const result = shell.echo(null);
+  const stdout = mocks.stdout();
+  const stderr = mocks.stderr();
+  t.falsy(shell.error());
+  t.is(result.code, 0);
+  t.is(stdout, 'null\n');
+  t.is(stderr, '');
 });
 
-test.cb('simple test with silent(true)', t => {
-  const script = 'require(\'../global.js\'); config.silent=true; echo(555);';
-  utils.runScript(script, (err, stdout) => {
-    t.falsy(err);
-    t.is(stdout, '555\n');
-    t.end();
-  });
+test.skip('simple test with silent(true)', t => {
```

Comment:
I think because this is testing with `config.silent = true;`, but we already have set `shell.config.silent = true` above on line 7, so this test is redundant.